[Cap-142-143-144] main menu

Kayram2710 opened the pull request:
# Goal
The goal of this PR is to bring a main menu into the application

# Key features
1. Added a menu UI in its own XML file
2. Created functionality to open and close main menu
3. Refactor of the maps_activity.xml to work on includes to modulate UI components

# Blue Print
![image](https://github.com/user-attachments/assets/f03a4cb4-d6d1-4d7b-baf7-def24e69165a)

# Result
<img src="https://github.com/user-attachments/assets/23c9d709-60c0-4c0f-afab-0da12f1b7236" alt="Bug" width="220"/>

![image](https://media4.giphy.com/media/v1.Y2lkPTc5MGI3NjExMW54cXN0ODlycXhzd2Jzc3doaWF1ZHBiY2xlaWg3OXJ1bnR2bjYxYyZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/3xGT5MVGw4zjaOzEmp/giphy.gif)

# Comments and Future Developement
1. The next step is to add a flavor for the UI clearing when opening the menu
2. Maybe making the buttons larger and perhpas the main menu slide larger to cover more of the screen.

Kayram2710 committed:
Setup Basic UI Layout

Kayram2710 committed:
Added Completed UI

Next going to add functionality

Kayram2710 committed:
Added Menu Functionality

JeanNaima commented on app/src/main/res/layout/activity_maps.xml:
Greate usage of layouts!

JeanNaima commented:
I noticed that our layouts are currently using "dp" to specify sizes which is not the best practice since the scaling will differ from phone to phone. Ideally we should use `android:layout_width="0dp"android:layout_height="0dp"` and use `app:layout_constraintWidth_percent="0.7" app:layout_constraintHeight_percent="1.0"` as an example. This doesn't have to be covered in this PR but its certainly a refactoring that we should keep in mind.

JeanNaima commented:
I also found some undesirable outcome that should be fixed. When the main menu is open, tapping on it will actually click on the map behind it and set a route. This can be probably be fixed by setting `android:clickable="true"` and `android:focusable="true"` on the sliding_menu.

### To reporduce

https://github.com/user-attachments/assets/193fe411-9f67-4d53-8250-e76eef3f4bb8

Kayram2710 commented:
Just fixed the bug addressed by @JeanNaima.

Kayram2710 committed:
Fixed Bug

Made screens unpassable

Kayram2710 committed:
Modulated Further To enable menu implementation amongst multiple activities

Created a menu controller that takes in layout parameters, this allows modulation and allows the menu to easily be implemented across multiple activities.

zachariyaJaveri reviewed (approved):
Good refactor, everything is addressed, I think this is ready for a merge
